server/src/main/kotlin/app/otakureader/server/service/SyncService.kt

```
@@ -0,0 +1,67 @@
+     * @return Pair of (data, timestamp) or null if no snapshot exists
+     */
+    fun getSnapshot(): Pair<String, Long>? {
+        if (!snapshotFile.exists() || !timestampFile.exists()) {
+            return null
+        }
+        
+        return try {
+            val data = snapshotFile.readText()
+            val timestamp = timestampFile.readText().toLongOrNull() ?: System.currentTimeMillis()
+            Pair(data, timestamp)
+        } catch (e: IOException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)